### PR TITLE
[BUGFIX] `argilla frontend`: redirect after login

### LINKED ISSUE #5633: [BUG-UI/UX] Cannot share dataset URLs after user signin on HF spaces
**Describe the bug**

When running the Argilla server in the HF spaces, sharing URLs with users needing an authentication flow does not work. After the user is authenticated, they are redirected to the home page instead of reaching the shared URL. This is working when running the Argilla server using the server docker image.

**To Reproduce**

The URL https://argilla-argilla-template-space.hf.space/dataset/3349bf8d-0755-403c-9964-a2e30544b97a/annotation-mode should be shown after sign-in

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**

 - OS [e.g. iOS]:
 - Browser [e.g. chrome, safari]:
 - Argilla Version [e.g. 1.0.0]:
 - ElasticSearch Version [e.g. 7.10.2]:
 - Docker Image (optional) [e.g. argilla:v1.0.0]:

**Additional context**

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

Closes https://github.com/argilla-io/argilla/issues/5633

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-frontend/v1/domain/usecases/oauth-login-usecase.test.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ describe("OAuthLoginUseCase should", () => {
 
   test("call the backend after logout", async () => {
     const auth = mock<IAuthService>();
+
```
Comment (frascuchon) [inline argilla-frontend/v1/domain/usecases/oauth-login-usecase.test.ts]:
```suggestion
```

File: argilla-frontend/pages/oauth/_provider/useOAuthViewModel.ts [hunk 2]
```diff
@@ -16,19 +16,28 @@ export const useOAuthViewModel = () => {
     await tryLogin();
   });
 
+  const redirect = () => {
+    let { redirect } = router.getQuery();
```
Comment (frascuchon) [inline argilla-frontend/pages/oauth/_provider/useOAuthViewModel.ts]: The Huggingface callback request does not send back the params. We must store the redirect URLs in the local/session storage cc @damianpumar

File: .github/workflows/argilla.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
   build:
     services:
       argilla-server:
-        image: argilladev/argilla-hf-spaces:develop
+        image: argilladev/argilla-hf-spaces:pr-5572
```
Comment (frascuchon) [inline .github/workflows/argilla.yml]:
```suggestion
        image: argilladev/argilla-hf-spaces:develop
```

File: argilla/CHANGELOG.md [hunk 4]
```diff
@@ -32,6 +32,10 @@ These are the section headers that we use:
 
 - Removed name sanitizing for dataset settings names. This may cause issues with old server versions. Especially when working with `from_hub`. ([#5574](https://github.com/argilla-io/argilla/pull/5574))
 
+### Removed
+
+- Removed name sanitizing for dataset settings names. This may cause issues with old server versions. Especially when working with `from_hub`. ([#5574](https://github.com/argilla-io/argilla/pull/5574))
+
```
Comment (frascuchon) [inline argilla/CHANGELOG.md]:
```suggestion
```

File: argilla-server/pdm.lock [hunk 5]
```diff
@@ -1775,6 +1775,62 @@ files = [
...
+    {file = "pillow-10.4.0-pp310-pypy310_pp73-win_amd64.whl", hash = "sha256:0755ffd4a0c6f267cccbae2e9903d95477ca2f77c4fcf3a3a09570001856c8a5"},
+    {file = "pillow-10.4.0-pp39-pypy39_pp73-macosx_10_15_x86_64.whl", hash = "sha256:a02364621fe369e06200d4a16558e056fe2805d3468350df3aef21e00d26214b"},
+    {file = "pillow-10.4.0-pp39-pypy39_pp73-macosx_11_0_arm64.whl", hash = "sha256:1b5dea9831a90e9d0721ec417a80d4cbd7022093ac38a568db2dd78363b00908"},
+    {file = "pillow-10.4.0-pp39-pypy39_pp73-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:9b885f89040bb8c4a1573566bbb2f44f5c505ef6e74cec7ab9068c900047f04b"},
+    {file = "pillow-10.4.0-pp39-pypy39_pp73-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:87dd88ded2e6d74d31e1e0a99a726a6765cda32d00ba72dc37f0651f306daaa8"},
+    {file = "pillow-10.4.0-pp39-pypy39_pp73-manylinux_2_28_aarch64.whl", hash = "sha256:2db98790afc70118bd0255c2eeb465e9767ecf1f3c25f9a1abb8ffc8cfd1fe0a"},
+    {file = "pillow-10.4.0-pp39-pypy39_pp73-manylinux_2_28_x86_64.whl", hash = "sha256:f7baece4ce06bade126fb84b8af1c33439a76d8a6fd818970215e0560ca28c27"},
```
Comment (frascuchon) [inline argilla-server/pdm.lock]:
```suggestion
```

File: argilla-server/pdm.lock [hunk 6]
```diff
@@ -1775,6 +1775,10 @@ files = [
     {file = "pillow-11.0.0.tar.gz", hash = "sha256:72bacbaf24ac003fea9bff9837d1eedb6088758d41e100c1552930151f677739"},
 ]
 
+    {file = "pillow-10.4.0-pp39-pypy39_pp73-win_amd64.whl", hash = "sha256:cfdd747216947628af7b259d274771d84db2268ca062dd5faf373639d00113a3"},
+    {file = "pillow-10.4.0.tar.gz", hash = "sha256:166c1cd4d24309b30d61f79f4a9114b7b2313d7450912277855ff5dfd7cd4a06"},
+]
+
```
Comment (frascuchon) [inline argilla-server/pdm.lock]:
```suggestion
```

**Review by leiyre — APPROVED** [review]

File: argilla-frontend/middleware/route-guard.ts [hunk 7]
```diff
@@ -17,23 +17,26 @@
...
+
+const { set } = useLocalStorage();
 
 export default ({ $auth, route, redirect }: Context) => {
   const { isRunningOnHuggingFace } = useRunningEnvironment();
 
+  if (route.path.startsWith("/api/")) {
+    return;
+  }
+
```
Comment (frascuchon) [inline argilla-frontend/middleware/route-guard.ts]:
```suggestion
```

**Pull request merged**